### ISSUE TITLE
Iceberg fluxes

### ISSUE THREAD

**Issue by pedrocol** [pedrocol/basal_mom5-collaborative-project#25]
Merino et al., 2016 propose a runoff field to simulate the iceberg fluxes (http://neichin.github.io/personalweb/material/meltwater/). Should we include this in our simulations?

**Comment by adele157** [comment]
I think for now, let's just leave the iceberg (calving) flux at the coast.
Then later we can run a separate experiment where we do both basal melt at
depth and calving spread according to the iceberg melt distribution. Best
to do these changes one at a time though, so we can attribute changes.

**Comment by adele157** [comment]
What temperature should the iceberg flux have? 0C? Or should we do the same thing we do with Gade line formulation for the basal melt? i.e. Account extract the latent heat needed to melt the ice from the ocean.

**Comment by adele157** [comment]
We decided that the temperature input for the icebergs should also follow the Gade line.

Next we need to decide what depth distribution to add the iceberg melt into the ocean. A lot of people seem to assume an average 250m thickness for near-Antarctic icebergs, e.g. see [Barbat et al. 2019](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2019JC015205) and references therein.

If we assume a 250 m thickness, what then is the profile of melt? It is constant from 0-250m depth? Or surface intensified? Or depth intensified?

**Comment by adele157** [comment]
Depth distribution of melt profiles here for two large North Atlantic icebergs: [Figure 2 in Schild et al. 2021](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2020GL089765). Melt seems to be surface and bottom intensified, which would be support for using a constant melt profile in depth between 0-250m.

**Comment by pedrocol** [comment]
I performed the interpolation of Merino et al. 2016 data with a Python regridder. In the plot below we have the regridded data on the left and the original data on the right. We can see that the NEMO grid is masked in places where ACCESS-OM2 is not

![Screenshot from 2022-12-14 00-36-28](https://user-images.githubusercontent.com/23285319/207758454-252e1454-b899-4430-b9cb-393b3d90d939.png)

If we want to use this data we'll have to somehow attribute some values in those places that are masked.
To be done: check if the interpretation is conserving the total amount of FW flux (if not write a code that does that), check if we are masking areas where NEMO is not

**Comment by pedrocol** [comment]
I discussed this with Pierre Mathiot, he confirms that the NEMO grid does not include those portions of Filchner and Ross. He works with an extended version of the ORCA grid (eORCA) and uses the Lagrangian Iceberg model, he mentioned that he may be able to provide a climatological melt rate from the icebergs on an eORCA025 grid from a reference simulation of his own

**Comment by adele157** [comment]
So we remember: We decided to spread the iceberg melt evenly between 0-100m, unless anyone can come up with a better justified distribution.

**Comment by adele157** [comment]
Plan for creating the iceberg flux file:
1. Conservatively regrid the Merino iceberg flux onto the ACCESS-OM2-01 grid.
2. Check that this iceberg flux matches the existing JRA55 calving flux when integrated over the whole Southern Ocean. If not, then rescale the regridded iceberg flux so it matches JRA55.
3. Check if any of the regridded flux is under land in ACCESS-OM2-01. If yes, then move this runoff to the coast (conservatively).
4. For regions where the NEMO grid is land, but ACCESS-OM2-01 is ocean, stretch the latitudinal profile of the regridded iceberg flux on the shelf between the 1000m isobath and the NEMO coast across the full ACCESS-OM2-01 shelf width (conservatively).
5. Do a final check that the total sum of iceberg flux matches JRA55 calving.

**Comment by mmr0** [comment]
> 

I can invent cases for top-intensified (waves, stronger currents), bottom intensified (warmer water - if deep enough) and (relatively) evenly distributed (side-wall double-diffusive layering) melting.. so .. I think even distribution is a good bet. I don't have any specific papers/studies in mind though sorry!

**Comment by adele157** [comment]
>  I can invent cases for top-intensified (waves, stronger currents), bottom intensified (warmer water - if deep enough) and (relatively) evenly distributed (side-wall double-diffusive layering) melting.. so .. I think even distribution is a good bet. I don't have any specific papers/studies in mind though sorry!

Yes, plus there's the consideration of how much melting occurs when icebergs are large (and deep) vs when they break up into smaller bergs (and are shallower). It seems like tracking the depth distribution of melt over the lifetime of icebergs is a research project waiting to be done!

**Comment by claireyung** [comment]
Linearly interpolated Merino flux: `/g/data/x77/cy8964/Post_Process/Merino_iceberg_flux_dist_interpolated.nc`

(Notebook modified slightly from Pedro, [here](https://github.com/pedrocol/basal_mom5-collaborative-project/blob/claire/notebooks/ICB/claire-icb.ipynb), to regrid the flux (per unit area) rather than the mass in a cell. Note that regridder is linear so doesn't account for different-sized cells - need the conservative version to work to do that!

Also not entirely sure if the longitude boundaries are treated quite right as the xesmf regridder crashes when I choose `periodic = True`.

**Comment by hrsdawson** [comment]
Conservatively (and conservative normed) regridded iceberg fluxes are also available here: `/g/data/e14/hd4873/model_data/access-om2/basal_melt_MOM5/`. These are comparable to the bilinearly interpolated fluxes (reproduces ~99% of the Merino flux). Why it's not 100, I'm not sure. `xesmf` was also stubbornly ignoring the `periodic=True` flag for me, perhaps this contributes to the discrepancy. 

All methods result in some fluxes on land in access - see figs at the end of [this notebook](https://github.com/pedrocol/basal_mom5-collaborative-project/blob/hannah/notebooks/ICB/icb_regridding_HD.ipynb). There's an option to feed in a land mask but when I tried this the regridded product reproduced less of the original fluxes (~98%). It seems like `xesfm` is not redistributing the masked portion of the flux to the neighbouring cells.

**Comment by adele157** [comment]
The iceberg flux input file is ready to run! It's here:
`/g/data/v45/akm157/model_data/Nacho_iceberg_fluxes/Merino_adjusted_fluxes.nc`

It was made with [this notebook](https://github.com/pedrocol/basal_mom5-collaborative-project/blob/main/notebooks/ICB/iceberg_flux_adjustments.ipynb). Not the prettiest code, but it satisfies everything we need:
* New iceberg flux input exactly matches the old JRA55 calving flux (`/g/data/ik11/inputs/JRA-55/RYF/v1-4/RYF.licalvf.1990_1991.nc`), when summed over area and time.
* Fluxes extend to the very south of the Ross and Weddell Seas. [There is a slight discontinuity at 75S, where I stretched the fluxes further south, but I think it's still a vast improvement on the original Merino dataset which didn't go far enough south.]
* No flux over land.

**Comment by PaulSpence** [comment]
Awesome!
P

**Comment by adele157** [comment]
From discussions today: make a plot comparing the longitudinal distribution of the JRA calving vs the iceberg flux input. Bin this onto perhaps 3 deg longitude bins so it's easier to compare them.
